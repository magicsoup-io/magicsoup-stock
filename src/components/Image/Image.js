import styled from 'styled-components'
import {
  height,
  borderRadius,
} from 'styled-system'

import Box from '../Box'

const themed = key => props => props.theme[key]

const Image = styled(Box)`
  maxWidth: 100%;
  height: auto;
  ${height}
  ${borderRadius}
  ${themed('Image')}
`

/** @component */
export default Image

Image.propTypes = {
  ...height.propTypes,
  ...borderRadius.propTypes,
}

Image.defaultProps = {
  as: 'img',
  m: 0,
}
