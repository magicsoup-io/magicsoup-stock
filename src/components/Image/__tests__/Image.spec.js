import React from 'react'
import renderer from 'react-test-renderer'

import Image from '../Image'

it('renders <Image /> without throwing', () => {
  const tree = renderer.create(
    <Image src="https://source.unsplash.com/random/1280x720" />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
