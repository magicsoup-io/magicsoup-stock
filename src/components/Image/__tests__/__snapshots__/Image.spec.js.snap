// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Image /> without throwing 1`] = `
<img
  className="sc-bdVaJa sc-gzVnrw gWAdUb"
  src="https://source.unsplash.com/random/1280x720"
/>
`;
